### PR TITLE
EVM-665 No Log Bloom

### PR DESCRIPTION
**Pull request by igorcrevar**
# Description

The log query will consume too much CPU and I/O especially with many DAPPs query all together.

Demo
Transactions with a log in block never get a bloom filter.

Transaction:
```
 [Transaction 0xe46e151d633637dee3425712b81935ab933b54c489eaa9fd630b3a83f75f62aa - Polygon EdgeNet Explorer](https://explorer-edgenet.polygon.technology/tx/0xe46e151d633637dee3425712b81935ab933b54c489eaa9fd630b3a83f75f62aa/logs) 
 ```
Query: `curl -H 'Content-Type: application/json' -X POST `
```
https:[//rpc-edgenet.polygon.technology](https://rpc-edgenet.polygon.technology/) -d '{"id": 0, "jsonrpc":"2.0", "method": "eth_getBlockByNumber", "params": ["8163744", false]}'
```

# Changes include

- [X] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [X] I have assigned this PR to myself
- [X] I have added at least 1 reviewer
- [X] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [X] I have tested this code with the official test suite
- [ ] I have tested this code manually

### CODE REVIEW

File: consensus/polybft/block_builder_test.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		Params: &chain.Params{
+			ChainID: chainID,
+			Forks:   forks,
+		},
+	}
+
+	stateStorage, err := itrie.NewLevelDBStorage(filepath.Join(tmpDir, "trie"), logger)
+	require.NoError(t, err)
+
+	mstate := itrie.NewState(stateStorage)
```
Comment (Stefan-Ethernal) [inline consensus/polybft/block_builder_test.go]: Maybe it is easier (and slightly faster) creating in memory storage instance?
Also in that case you don't need `tmpDir` variable (BB Show can be retired 🙂 )
```suggestion
	mstate := itrie.NewState(itrie.NewMemoryStorage())
```

File: consensus/polybft/block_builder_test.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	"github.com/0xPolygon/polygon-edge/state"
+	itrie "github.com/0xPolygon/polygon-edge/state/immutable-trie"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBlockBuilder_BuildBlockTxOneFailedTxAndOneTakesTooMuchGas(t *testing.T) {
+	const (
```
Comment (Stefan-Ethernal) [inline consensus/polybft/block_builder_test.go:24]: Nit:
```suggestion
        t.Parallel()
        
	const (
```

File: consensus/polybft/block_builder_test.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+		}
+	}
+
+	balanceMap := map[types.Address]*chain.GenesisAccount{}
+
+	for i, acc := range accounts {
+		// the third tx will fail because of insufficient balance
+		if i != 2 {
+			balanceMap[types.Address(acc.Ecdsa.Address())] = &chain.GenesisAccount{
+				Balance: big.NewInt(1_000_000_000_000),
```
Comment (Stefan-Ethernal) [inline consensus/polybft/block_builder_test.go]: If value can be arbitrary here, maybe consider using `ethgo.Ether` or `ethgo.Gwei` function here (for the sake of better readability). WDYT?
```suggestion
				Balance: ethgo.Ether(1),
```

**Review by Stefan-Ethernal — APPROVED** [review]

File: consensus/polybft/block_builder_test.go (hunk 3 above)
Comment (igorcrevar) [reply consensus/polybft/block_builder_test.go]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1550/commits/1d61f96bb3d41b1066ea6ddb6dcb7a2804008ab7

File: consensus/polybft/block_builder_test.go (hunk 2 above)
Comment (igorcrevar) [reply consensus/polybft/block_builder_test.go:24]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1550/commits/1d61f96bb3d41b1066ea6ddb6dcb7a2804008ab7

File: consensus/polybft/block_builder_test.go (hunk 1 above)
Comment (igorcrevar) [reply consensus/polybft/block_builder_test.go]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1550/commits/1d61f96bb3d41b1066ea6ddb6dcb7a2804008ab7

**Review by goran-ethernal — APPROVED** [review]

**Pull request merged**